src/components/FeatureViews/Scheduler/SessionView.js

```diff
@@ -31,36 +41,109 @@ class SessionView extends Component {
+                    id: 4,
+                },
+                3: {
+                    name: "Able Strong Body",
+                    id: 11,
+                }
+            }
         };
     }
 
     handleOpen = () => {
-        this.setState({"open": true});
+        this.setState({ "open": true });
     }
 
     handleClose = () => {
-        this.setState({"open": false});
+        this.setState({ "open": false });
     }
 
-    componentWillMount() {
-        this.setState(() => {
-            const sessionData = this.props.courseSessions[this.props.computedMatch.params.course_id][this.props.computedMatch.params.session_id];
-            const courseData = this.props.courses[this.props.computedMatch.params.course_id];
-            return {
-                sessionData,
-                courseData,
-            };
-        });
+    componentDidMount() {
+        this.props.calendarActions.fetchSessions({ id: this.props.match.params.session_id });
+        this.props.courseActions.fetchCourses(this.props.match.params.course_id);
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (this.props !== prevProps &&
+            this.props.courseSessions.length !== 0 &&
+            Object.entries(this.props.courses).length !== 0) {
+            this.setState(() => {
+                const sessionData = this.props.courseSessions.find((session) => {
+                    return session.course === Number(this.props.match.params.course_id) &&
+                        session.id === Number(this.props.match.params.session_id);
+                });
+                sessionData["start"] = new Date(sessionData.start_datetime)
+                    .getDay()
+                // let sessionTime = sessionData.start.substring(sessionData.start.indexOf(" "));
+
+                // sessionData["start"] = sessionData.start.replace(sessionTime, " | " + parseTime(new Date(sessionData.start_datetime).toUTCString()));
+
+                const startTime = parseTime(new Date(sessionData.start_datetime).toUTCString());
+                const endTime = parseTime(new Date(sessionData.end_datetime).toUTCString());
+                sessionData["startTime"] = startTime;
+                sessionData["endTime"] = endTime;
+
+
+                let courseData = this.props.courses[this.props.match.params.course_id];
+                if (courseData && !this.props.requestStatus["instructor"][GET][courseData.instructor_id]) {
+                    this.props.userActions.fetchInstructors(courseData.instructor_id);
+                }
+                return {
+                    sessionData,
+                    courseData,
+                };
+            });
+
+        }
     }
 
+
+
+
     render() {
+        let instructor = this.state.courseData && this.props.instructors[this.state.courseData.instructor_id] ? this.props.instructors[this.state.courseData.instructor_id] : { name: "N/A" };
+        let dayConverter = {
```

Comment:
This can be moved outside the component as a constant since it does not depend on state, props, or Redux.